python/arcticdb/arctic.py

```diff
@@ -40,7 +40,7 @@ class Arctic:
     _created_lib_names: Optional[List[str]] = None
     _accessed_libs: Optional[List[NativeVersionStore]] = None
 
-    def __init__(self, uri: str, encoding_version: EncodingVersion = DEFAULT_ENCODING_VERSION):
+    def __init__(self, uri: str, encoding_version: EncodingVersion = DEFAULT_ENCODING_VERSION, credential: AzureDefaultCredential = None):
```

Comment:
Wonder if we are at the end of the road with with `Arctic` constructor, and should introduce a new mode like,

```
Arctic(AzureStorage(endpoint=xxx, credential=AzureCredential(xxx), ...))
```

(while keeping the URI mode for backwards compat)?

What do you think?